File: rails/app/services/content/draft_session_service.rb
module Content
  class DraftSessionService
    def initialize(repository = DraftSession)
      @repository = repository
    end

    def create(attributes)
      raise NotImplementedError, "#{self.class}#create not implemented"
    end

    def update(entity, attributes)
      raise NotImplementedError, "#{self.class}#update not implemented"
    end

    def start(id)
      instance = DraftSession.find(id)
      instance.start()
      instance.save!
    end

    def abandon(id)
      instance = DraftSession.find(id)
      instance.abandon()
      instance.save!
    end

    def complete(id)
      instance = DraftSession.find(id)
      instance.complete()
      instance.save!
    end

    def is_full(id)
      instance = DraftSession.find(id)
      result = instance.is_full()
      instance.save!
      result
    end
  end
end

File: laravel/app/Services/Content/DraftSessionService.php
<?php

namespace App\Services\Content;

use App\Models\Content\DraftSession;

class DraftSessionService
{
    public function create(array $data): DraftSession
    {
        throw new \LogicException('Not implemented');
    }

    public function update(DraftSession $draftSession, array $data): DraftSession
    {
        throw new \LogicException('Not implemented');
    }
    public function start(int $id): void
    {
        $draftSession = DraftSession::findOrFail($id);
        $draftSession->start();
        $draftSession->save();
    }

    public function abandon(int $id): void
    {
        $draftSession = DraftSession::findOrFail($id);
        $draftSession->abandon();
        $draftSession->save();
    }

    public function complete(int $id): void
    {
        $draftSession = DraftSession::findOrFail($id);
        $draftSession->complete();
        $draftSession->save();
    }

    public function isFull(int $id): bool
    {
        $draftSession = DraftSession::findOrFail($id);
        $result = $draftSession->isFull();
        $draftSession->save();
        return $result;
    }
}
